libraries/standard/mqtt/src/mqtt.c

```diff
@@ -214,6 +245,107 @@ static MQTTStatus_t sendPublish( const MQTTContext_t * const pContext,
 
 /*-----------------------------------------------------------*/
 
+static uint32_t calculateElapsedTime( uint32_t later,
+                                      uint32_t start )
+{
+    return later - start;
+}
+
+/*-----------------------------------------------------------*/
+
+static MQTTStatus_t receivePacket( MQTTContext_t * const pContext,
+                                   MQTTPacketInfo_t incomingPacket,
+                                   uint32_t remainingTimeMs )
+{
+    return MQTTSuccess;
+}
+
+/*-----------------------------------------------------------*/
+
+static MQTTStatus_t receiveConnack( MQTTContext_t * const pContext,
+                                    uint32_t timeoutMs,
+                                    MQTTPacketInfo_t * const incomingPacket,
+                                    bool * const pSessionPresent )
+{
+    MQTTStatus_t status = MQTTSuccess;
+    MQTTGetCurrentTimeFunc_t getTimeStamp = NULL;
+    uint32_t entryTime = 0U, remainingTimeMs = 0U, timeTaken = 0U;
+
+    assert( pContext != NULL );
+    assert( incomingPacket != NULL );
+    assert( pContext->callbacks.getTime != NULL );
+
+    getTimeStamp = pContext->callbacks.getTime;
+    /* Get the entry time for the function. */
+    entryTime = getTimeStamp();
+
+    do
+    {
+        /* Transport read for incoming CONNACK packet type and length. */
+        status = MQTT_GetIncomingPacketTypeAndLength( pContext->transportInterface.recv,
```

Comment:
I'd note that this is a blocking call so that the reader knows this isn't a busy waiting loop.

Reply:
Done.